### PR TITLE
Provide interface for hb_style and font variation metrics

### PR DESCRIPTION
**Pull request by zauguin**
Since HarfBuzz 3.0, hb-style is no longer experimental. It is a very useful interface to get some font metadata while automatically selecting between font variation based data if applicable and static fields otherwise.

Also adds a interface for `font_ot_metrics_get_variation` and friends to provide full access to font variation metrics if necessary.

### CODE REVIEW

**Comment by zauguin** [comment]
Currently `style_get_value` is only provided if the used HarfBuzz version is sufficiently new (at least 3.0.0).

The question is how to handle this for the tests. Currently the test runs unconditionally, so it fails if the HarfBuzz version is too old.(E.g. on GitHub Actions right now) The motivation for that is that the tests are mostly run during development, where hiding issues by automatically disabling the test seems worse than explicit errors when the version is too old to test all functionality. If we keep this, we should probably find a way to get a newer HarfBuzz version on for the Actions. 

An alternative would be to disable the tests for older HarfBuzz versions in order to allow testing to run on all systems according to the system specific capabilities.

**Comment by deepakjois** [comment]
Do you know why one of the tests is failing?

**Comment by deepakjois** [comment]
> An alternative would be to disable the tests for older HarfBuzz versions in order to allow testing to run on all systems according to the system specific capabilities.

I would prefer to keep the tests in, and find a way to run it only for the latest version. I can't remember now, but don't we have a way to query the version from the Harfbuzz API?

At some point we would be able to pull in the latest harfbuzz. Maybe the Ubuntu version of the container image is is too old? I can work on upgrading harfbuzz in the CI build sometime in late Nov when I might have some time.

**Comment by zauguin** [comment]
> I would prefer to keep the tests in, and find a way to run it only for the latest version.

Done.

> Maybe the Ubuntu version of the container image is is too old?

It's currently running on Ubuntu 20.04 which has HarfBuzz 2.6.4, but even the latest release (21.10) is still on HarfBuzz 2.7.4.

The next release (22.04) will be LTS and therefore will probably become available on GH Actions, so we can hope that that one will ship HarfBuzz 3.x. (Currently the development release is still on 2.7.4, but the feature freeze isn't until February. On the other hand I wouldn't be surprised if they don't merge the major release of HarfBuzz in their LTS version.)

So basically what I'm trying to say is: If we want to test with HarfBuzz 3, we probably have to build HarfBuzz by ourselves for now.

**Comment by deepakjois** [comment]
> So basically what I'm trying to say is: If we want to test with HarfBuzz 3, we probably have to build HarfBuzz by ourselves for now.

Got it. I suppose building Harfbuzz as part of the CI build is a possibility. But I am not sure if it's worth investing time in that given it's just for one feature (and one test). If we have more features that depend on harfbuzz v3, i could be convinced that it is more critical.

OTOH, if somebody feels like it's worth doing, I would happily merge the change.
